### ISSUE TITLE
[2.9 ACRA] media recording stop failure can cause a crash

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#5508]
This looks like a runtime crash we could catch and message better to the user?

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/23c723a7-e244-4ede-ba8b-f5988cdc31f3

```
java.lang.RuntimeException: stop failed.
at android.media.MediaRecorder.stop(Native Method)
at com.ichi2.anki.multimediacard.AudioView.notifyStopRecord(AudioView.java:2)
at com.ichi2.anki.multimediacard.AudioView$RecordButton$1.onClick(AudioView.java:4)
at android.view.View.performClick(View.java:5232)
at android.view.View$PerformClick.run(View.java:21258)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:179)
at android.app.ActivityThread.main(ActivityThread.java:5650)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:791)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:681)
at android.view.View.measure(View.java:17611)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5547)
at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1467)
at android.widget.LinearLayout.measureVertical(LinearLayout.java:747)
at android.widget.LinearLayout.onMeasure(LinearLayout.java:638)
at android.view.View.measure(View.java:17611)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5547)
at android.view.View.measure(View.java:17611)
at android.view.View.measure(View.java:17611)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5547)
at android.widget.FrameLayout.onMeasure(FrameLayout.java:436)
at android.view.View.measure(View.java:17611)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5547)
at android.widget.FrameLayout.onMeasure(FrameLayout.java:436)
at android.view.View.measure(View.java:17611)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5547)
at android.widget.FrameLayout.onMeasure(FrameLayout.java:436)
at android.view.View.measure(View.java:17611)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5547)
at android.view.View.measure(View.java:17611)
at android.view.LayoutInflater.rInflate(LayoutInflater.java:809)
```

**Comment by mikehardy** [comment]
The documentation for the stop call explicitly says this is possible, and it is how the API communicates that no valid data was received, so you have to catch it and in the handler clean up any data:

https://developer.android.com/reference/android/media/MediaRecorder#stop()

**Comment by mikehardy** [comment]
Fixed in #5791
